api/v1beta1/hcloudmachine_validation_test.go

```diff
@@ -129,22 +121,19 @@ func TestValidateHCloudMachineSpec(t *testing.T) {
 					PlacementGroupName: createPlacementGroupName("placement-group-1"),
 				},
 			},
-			want: field.ErrorList{},
+			want: nil,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got := validateHCloudMachineSpec(tt.args.oldSpec, tt.args.newSpec)
 
-			if len(got) != len(tt.want) {
-				t.Errorf("validateHCloudMachineSpec() = %v, want %v", got, tt.want)
-				return
-			}
-
-			for i, err := range got {
-				assert.Equal(t, tt.want[i].Type, err.Type)
-				assert.Equal(t, tt.want[i].Field, err.Field)
-				assert.Equal(t, tt.want[i].Detail, err.Detail)
+			if tt.want == nil {
+				assert.Empty(t, got)
+			} else if assert.Len(t, got, 1) {
```

Comment:
if "len(got) != 1" then you should fail right?

Reply:
yes